lightning/src/sign/mod.rs

```diff
@@ -776,6 +776,10 @@ pub trait ChannelSigner {
 	/// Returns the holder's channel public keys and basepoints.
 	///
 	/// This method is *not* asynchronous. Instead, the value must be cached locally.
+	///
+	/// NOTE: The [`ChannelPublicKeys::funding_pubkey`] returned will not necessarily correspond to
+	/// the channel's current holder public key, as it may have rotated due a splice. The value
+	/// should not be relied upon once the channel's initial funding transaction has been created.
```

Comment:
I wonder if we should return the `ChannelPublicKeys` as part of the `derive_channel_signer` so that we can remove this method. Then we wouldn't need this caveat. Doesn't need to be done in this PR if it's going to be involved, though.